pkg/rule/repo.go

```diff
@@ -0,0 +1,155 @@
+package rule
+
+import (
+	"time"
+
+	"github.com/lifesum/configsum/pkg/errors"
+)
+
+const (
+	kindOverride kind = iota + 1
+	kindExperiment
+	kindRollout
+)
+
+type kind uint8
+
+type parameters map[string]interface{}
+
+type bucket struct {
+	Name       string
+	Parameters parameters
+	Percentage int
+}
+
+type context struct {
+	user contextUser
+}
+type contextUser struct {
+	age uint8
+	id  string
+}
+
+// Repo provides access to rules.
+type Repo interface {
+	lifecycle
+
+	Create(input rule) (rule, error)
+	Get(configID, name string) (rule, error)
+	UpdateWith(input rule) (rule, error)
+	ListAll(configID string) ([]rule, error)
+	ListActive(configID string, now time.Time) ([]rule, error)
+}
+
+type rule struct {
+	active      bool
+	activatedAt time.Time
+	buckets     []bucket
+	configID    string
+	createdAt   time.Time
+	criteria    *criteria
+	description string
+	endTime     *time.Time
```

Comment:
Instead of pointers those could be just `time.Time` and when `IsZero` known to be not set.